Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -953,6 +954,166 @@ function Parse-Exception($exception){
     return $null
 }
 
+function Get-AzureNetworkInterfaceDetails
+{
+    [CmdletBinding()]
+    param([String] [Parameter(Mandatory = $true)] $resourceGroupName,
+          [Object] [Parameter(Mandatory = $true)] $endpoint)
+
+    $accessToken = Get-SpnAccessToken $endpoint
+    $subscriptionId = $endpoint.Data.SubscriptionId.ToLower()
+
+    Write-Verbose "[Azure Rest Call] Get Resource Groups"
+    
+    $method = "GET"
+    $uri = "$($endpoint.Url)/subscriptions/$subscriptionId/resourceGroups/$resourceGroupName/providers/Microsoft.Network/networkInterfaces?api-version=$azureStackapiVersion"
+    $headers = @{Authorization=("{0} {1}" -f $accessToken.token_type, $accessToken.access_token)}
+
+    $networkInterfaceDetails = (Invoke-RestMethod -Uri $uri -Method $method -Headers $headers -ContentType $script:jsonContentType)
```

Comment:
You should handle proxy. Look for instances where Proxy-Uri is being used.

Reply:
This comment is applicable to all instances where you are making Invoke-RestMethod call.